osquery/tables/system/darwin/unified_log.mm

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2021-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <Foundation/Foundation.h>
+#include <OSLog/OSLog.h>
+
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+
+#include <boost/algorithm/string/replace.hpp>
+
+namespace ba = boost::algorithm;
+
+namespace osquery {
+namespace tables {
+
+const std::map<ConstraintOperator, NSPredicateOperatorType> supportedOps = {
+    {EQUALS, NSEqualToPredicateOperatorType},
+    {GREATER_THAN, NSGreaterThanPredicateOperatorType},
+    {GREATER_THAN_OR_EQUALS, NSGreaterThanOrEqualToPredicateOperatorType},
+    {LESS_THAN, NSLessThanPredicateOperatorType},
+    {LESS_THAN_OR_EQUALS, NSLessThanOrEqualToPredicateOperatorType},
+    {LIKE, NSLikePredicateOperatorType}};
+
+const std::map<std::string, std::string> columnToOSLogEntryProp = {
```

Comment:
I'm having trouble finding docs on the object in question, but `man logs` also describes `eventType`, `messageType`, `processImagePath`, and `senderImagePath`. Do you think we care about any of those?

Reply:
the best place to find docs would probably be the [entry for OSLogEntryLog](https://developer.apple.com/documentation/oslog/oslogentrylog?language=objc). `processImagePath` and `senderImagePath` aren't exposed in this API. `eventType` might be something we can extract, I'll take a look, though I don't think so. `messageType` is the severity of the message when the event type is a log or trace event--I'm not sure it's exposed in this api either. I'll dig in.

Reply:
Something like this might work:

const char* aulLevelNames[] = {
    [OSLogEntryLogLevelUndefined] = "undefined",
    [OSLogEntryLogLevelDebug] = "debug",
    [OSLogEntryLogLevelInfo] = "info",
    [OSLogEntryLogLevelNotice] = "default",
    [OSLogEntryLogLevelError] = "error",
    [OSLogEntryLogLevelFault] = "fault",
};

      if ([entry isKindOfClass:[OSLogEntryLog class]]) { 
        r["type"] = aulLevelNames[[(OSLogEntryLog*)entry level]];
      } else if ([entry isKindOfClass:[OSLogEntryActivity class]]) {
         r["type"] = "activity";
      }

Reply:
I think this is slightly dangerous. If the set of logtypes change then this could fault. However, that seems unlikely. Can you use a C++ container to get this same effect, then you will except vs. fault.